packages/vscode-ui5-language-assistant/test/suite/test-utils.ts

```diff
@@ -0,0 +1,56 @@
+import { promises as fs } from "fs";
+import * as vscode from "vscode";
+import * as chai from "chai";
+import { expect } from "chai";
+import deepEqualInAnyOrder from "deep-equal-in-any-order";
+import { TextDocument } from "vscode-languageclient";
+
+chai.use(deepEqualInAnyOrder);
+
+export async function setXMLContent(
+  content: string,
+  xmlPath: string
+): Promise<void> {
+  await fs.writeFile(xmlPath, content);
+  await sleep(1000);
```

Comment:
an interesting question how long a sleep is needed here or if a sleep is even needed.
If you do `writeFileSync` would a sleep still be needed?

In general I would expect VSCode to pick up file changes in less than 1 second.
If the `sync` approach is unstable perhaps try smaller sleep timers as this `sleep` would be called many times?

Reply:
`sync` is unstable, but I set the sleep timer to 500ms.